src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Reload/ReloadableProject.cs

```diff
@@ -22,18 +21,22 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS
     [AppliesTo("HandlesOwnReload")]
     internal class ReloadableProject : OnceInitializedOnceDisposedAsync, IReloadableProject
     {
+        private readonly IUnconfiguredProjectVsServices _projectVsServices;
+        private readonly IProjectReloadManager _reloadManager;
+        private readonly IProjectReloadInterceptor _projectReloadInterceptor;
 
         [ImportingConstructor]
-        public ReloadableProject(IUnconfiguredProjectVsServices projectVsServices, IProjectReloadManager reloadManager)
+        public ReloadableProject(
+            IUnconfiguredProjectVsServices projectVsServices,
+            IProjectReloadManager reloadManager,
+            [Import(AllowDefault = true)]IProjectReloadInterceptor projectReloadInteceptor = null)
```

Comment:
Should make this an `OrderPrecedenceImportCollection` if we're going to import something, so other extensions can register reasons to stop reload.